virtcontainers/network.go

```diff
@@ -593,6 +694,10 @@ func initNetworkCommon(config NetworkConfig) (string, bool, error) {
 		return path, true, nil
 	}
 
+	if hostNetworkingRequested(config.NetNSPath) {
```

Comment:
Move it before `createNetNS()`? No need to check the same ns created by us. Otherwise `createNetNS()` is the broken one.

Reply:
@bergwolf I didnt quite get you. `createNetNS` creates the netns is no netns is provided and returns immediately.

Reply:
Hi @amshinde - I think he's saying do something like this:

```go
if config.NetNSPath != "" {
        if hostNetworkingRequested(config.NetNSPath) {
                return "", false, fmt.Errorf("Host networking requested, not supported by runtime")
        }
} else if config.NetNSPath == "" {
        path, err := createNetNS()
        if err != nil {
                return "", false, err
        }

        return path, true, nil
}
```

The reason being that the runtime itself wouldn't be creating a host ns (via `createNetNS()`) so why call `hostNetworkingRequested()` unconditionally?

Reply:
@jodh-intel The `hostNetworkingRequested` path will run only if no net namespace is provided.
I had something like this before
```
   if config.NetNSPath == "" {
            path, err := createNetNS()
            if err != nil {
                   return "", false, err
            }
           return path, true, nil
   } else {
            if hostNetworkingRequested(config.NetNSPath) {
                  return "", false, fmt.Errorf("Host networking requested, not supported by runtime")
            }
   }
```

But golint specified that I take out the else since all code paths in the if block  return.

Reply:
Ah - sorry. I'd missed that 2nd `return` when `NetNsPath == ""`.